### PR TITLE
applications: asset_tracker_v2: update config handling for nRF cloud devices

### PR DESCRIPTION
**Pull request by jayteemo**
Updated nrf_cloud lib to allow application to send shadow updates earlier in connection process.
Updated nrf_cloud lib to send shadow data to app even if no config section is present.
Updated ATv2 to allow config handling for nRF Cloud builds prior to "connected" state.
CIA-547

Included a commit for some nrf_cloud lib cleanup.
CIA-546 

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

### CODE REVIEW

**Comment by jayteemo** [comment]
for nrf cloud, when a new device is added there is no config section in the shadow.
this was noticed when testing the new config card on nrf cloud.

not sure if this change would affect other cloud backends, so i added the ifdef around the actual config send.
let me know if you have a better way to solve this.

**Comment by simensrostad** [comment]
> for nrf cloud, when a new device is added there is no config section in the shadow. this was noticed when testing the new config card on (dev) nrf cloud.
> 
> not sure if this change would affect other cloud backends, hence the ifdef around the actual config send. let me know if you have a better way to solve this.

This can be aligned with the other cloud integrations by updating the nRF library to always forward the content of the device shadow to the application. Then the application can determine if any of the content in the shadow is of interest and decide to act on that. This would also accommodate for applications that uses different naming/structures in the device shadow. (Right now the shadow state is only forwarded if `config` if I'm not mistaken?

In the ATv2, If the device shadow/twin does not contain any configuration values (parsed in the app codec), we explicitly act on that by sending all the current configuration values.

**Comment by jayteemo** [comment]
Correct only config gets sent.
I initially did update the nrf_cloud lib to send the full shadow but since the cloud is not in a fully connected state, atv2 fails to send the config.

**Comment by jayteemo** [comment]
@simensrostad I updated nrf_cloud lib to always send shadow data and to allow shadow updates earlier in the connection process.

**Review by simensrostad — APPROVED** [review]
LGTM

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -59,7 +59,9 @@ This section provides detailed lists of changes by :ref:`application <applicatio
 nRF9160: Asset Tracker v2
 -------------------------
 
-|no_changes_yet_note|
+* Updated:
+
+  * For nRF Cloud builds, the configuration section in the shadow will now be initialized during the cloud connection process.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:64]:
```suggestion
  * For nRF Cloud builds, the configuration section in the shadow is now initialized during the cloud connection process.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -96,6 +98,16 @@ Libraries
...
 
+Libraries for networking
+------------------------
+
+  * :ref:`lib_nrf_cloud` library:
+
+    * Updated:
+
+      * During the connection process, shadow data is sent to the application even if no "config" section is present.
+      * The application can now send shadow updates earlier in the connection process.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:109]:
```suggestion
* :ref:`lib_nrf_cloud` library:

  * Updated:

     * During the connection process, shadow data is sent to the application even if no "config" section is present.
     * The application can now send shadow updates earlier in the connection process.
```
Just minor alignments

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Comment by carlescufi** [comment]
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-6900
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**